rapid7_insight_agent/plugin.spec.yaml

```diff
@@ -0,0 +1,222 @@
+tags:
+- rapid7
+- agent
+- insight
+- edr
+- endpoint
+- detection
+- response
+- idr
+- ivm
+hub_tags:
+  use_cases: [threat_detection_and_response]
+  keywords: [rapid7, agent, endpoint, response, detection, insight, idr, ivm]
+  features: []
+types:
+  attribute:
+    key:
+      title: "Key"
+      type: string
+      description: "Key"
+      required: false
+    value:
+      title: "Value"
+      type: string
+      description: "Value"
+      required: false
+  hostName:
+    name:
+      title: "Name"
+      type: string
+      description: "Name"
+      required: false
+  primaryAddress:
+    ip:
+      title: "IP Address"
+      type: string
+      description: "IP address"
+      required: false
+    mac:
+      title: "MAC Address"
+      type: string
+      description: "MAC address"
+      required: false
+  uniqueIdentity:
+    id:
+      title: "ID"
+      type: string
+      description: "ID"
+      required: false
+    source:
+      title: "Source"
+      type: string
+      description: "Source"
+      required: false
+  agent:
+    attributes:
+      title: "Attributes"
+      type: "[]attribute"
+      description: "Attributes"
+      required: false
+    description:
+      title: "Description"
+      type: string
+      description: "Description"
+      required: false
+    hostNames:
+      title: "Host Names"
+      type: "[]hostName"
+      description: "Host names"
+      required: false
+    id:
+      title: "ID"
+      type: string
+      description: "ID"
+      required: false
+    primaryAddress:
+      title: "Primary Address"
+      type: primaryAddress
+      description: "Primary address"
+      required: false
+    uniqueIdentity:
+      title: "Unique Identity"
+      type: "[]uniqueIdentity"
+      description: "Unique identity"
+      required: false
+    vendor:
+      title: "Vendor"
+      type: string
+      description: "Vendor"
+      required: false
+    version:
+      title: "Version"
+      type: string
+      description: "Version"
+      required: false
+
+connection:
+  api_key:
+    title: API Key
+    description: User or Organization Key from the Insight Platform
+    type: credential_secret_key
+    example: a5zy0a6g-504e-46bz-84xx-1b3f5ci36l99
+    required: true
+  org_id:
+    title: Organization ID
+    description: Organization ID
+    type: credential_secret_key
+    example: 2cg99z6y-795n-7bzf-hj67-12355h554974
+    required: true
+  region:
+    title: Region
+    description: Region
+    type: string
+    default: United States
+    example: United States
+    required: true
+    enum:
+      - United States
+      - Europe
+      - Canada
+      - Australia
+      - Japan
+actions:
+  get_agent_details:
+    title: Get Agent Details
+    description: This action is used to find and display detailed information about a device
+    input:
+      agent:
+        title: Agent
+        description: IP address, MAC address, or hostname of the device to get information from
```

Comment:
We should allow the agent ID here too to be consistent with the other actions

Reply:
Can't do it. You can't distinguish between host names and UID.